src/kaocha/plugin/alpha/spec_test_check.clj

```diff
@@ -1,40 +1,177 @@
-           (update test assoc ::stc/opts opts)
+           (meta-merge (::global stc-configs) test (::cli stc-configs))
            test))
        tests))
 
-(defn ^:private default-test-suite [{::stc/keys [opts] :as config}]
-  {:kaocha.testable/type        :kaocha.type/spec.test.check
-   :kaocha.testable/id          :generative-fdef-checks
-   :kaocha.filter/skip-meta     [:kaocha/skip :no-gen]
-   :kaocha/source-paths         ["src"],
-   :kaocha.spec.test.check/syms :all-fdefs
-   ::stc/opts                   opts})
+(s/fdef extract-stc-config
+  :args (s/cat :config         :kaocha/config
+               :config-mapping ::config-mapping)
+  :ret  :kaocha/config)
+
+(defn ^:private extract-stc-config
+  [config config-mapping]
+  (reduce-kv (fn [acc source-path dest-path]
+               (if-some [setting (get-in config source-path)]
+                 (assoc-in acc dest-path setting)
+                 acc))
+             {}
+             config-mapping))
+
+(s/fdef extract-stc-configs
+  :args (s/cat :config :kaocha/config)
+  :ret  ::stc-configs)
+
+(defn ^:private extract-stc-configs
+  [config]
+  (reduce-kv (fn [acc key config-mapping]
+               (assoc acc key (extract-stc-config config config-mapping)))
+             {}
+             config-mappings))
+
+(s/fdef override-stc-config
+  :args (s/cat :config :kaocha/config)
+  :ret  :kaocha/config)
+
+(defn ^:private override-stc-config
+  [config]
+  (let [stc-configs (extract-stc-configs config)]
+    (-> config
+        (meta-merge (::cli stc-configs))
+        (update :kaocha/tests override-test-stc-config stc-configs))))
+
+(s/fdef ensure-stc-suite
+  :args (s/cat :tests (s/nilable ::tests))
+  :ret  ::tests)
 
-(defn ^:private override-stc-settings [config]
-  (assoc config :kaocha/tests (tests-with-overridden-stc-opts config)))
+(defn ^:private ensure-stc-suite
+  [tests]
+  (if (some is-stc? tests)
+    tests
+    (conj tests default-stc-suite)))
 
-(defn ^:private add-default-test-suite [config]
-  (update config :kaocha/tests conj (default-test-suite config)))
+(def ^:private extract-stc-suite-ids-xform
+  (comp (filter is-stc?)
+        (map :kaocha.testable/id)))
+
+(s/fdef stc-suite-ids
+  :args (s/cat :config :kaocha/config)
+  :ret  (s/coll-of keyword? :kind set?))
+
+(defn ^:private stc-suite-ids
+  [config]
+  (into #{} extract-stc-suite-ids-xform (:kaocha/tests config)))
+
+(s/fdef stc-suites-not-selected?
+  :args (s/cat :config :kaocha/config)
+  :ret  boolean?)
+
+(defn ^:private stc-suites-not-selected?
+  [{:kaocha/keys [cli-args] :as config}]
+  (boolean (and (seq cli-args)
+                (not-any? (stc-suite-ids config) cli-args))))
+
+(s/fdef config-hook
+  :args (s/cat :config :kaocha/config)
+  :ret  :kaocha/config)
+
+(defn ^:private config-hook
+  [config]
+  (let [config-with-suite (update config :kaocha/tests ensure-stc-suite)]
+    (if (stc-suites-not-selected? config-with-suite)
+      config
+      (override-stc-config config-with-suite))))
```

Comment:
`fdef` can be a little verbose. Orchestra's [defn-spec](https://github.com/jeaye/orchestra#defn-spec) is some nice sugar around it—thoughts?

Reply:
Honestly I don't use spec and fdef that much so no strong opinions. Since we already have orchestra as a dependency I think it's fine to use it.